infra/azure/main.tf

```diff
@@ -60,18 +62,50 @@ resource "azurerm_kubernetes_cluster" "vdc" {
 }
 
 resource "azurerm_kubernetes_cluster_node_pool" "vdc_pool" {
-  name = "pool"
+  name                  = "pool"
   kubernetes_cluster_id = azurerm_kubernetes_cluster.vdc.id
-  vm_size = "Standard_D2_v2"
-  vnet_subnet_id = azurerm_subnet.k8s_subnet.id
+  vm_size               = "Standard_D2_v2"
+  vnet_subnet_id        = azurerm_subnet.k8s_subnet.id
 
   enable_auto_scaling = true
 
-  node_count = 1
   min_count = 0
   max_count = 200
 }
 
+resource "azurerm_public_ip" "gateway_ip" {
+  name                = "gateway-ip"
+  resource_group_name = azurerm_kubernetes_cluster.vdc.node_resource_group
+  location            = data.azurerm_resource_group.rg.location
+  sku                 = "Standard"
+  allocation_method   = "Static"
+}
+
+resource "azurerm_container_registry" "acr" {
+  name                = var.acr_name
+  resource_group_name = data.azurerm_resource_group.rg.name
+  location            = data.azurerm_resource_group.rg.location
+  sku                 = var.acr_sku
+}
+
+resource "azurerm_role_assignment" "vdc_to_acr" {
+  scope                = azurerm_container_registry.acr.id
+  role_definition_name = "AcrPull"
+  principal_id         = azurerm_kubernetes_cluster.vdc.kubelet_identity[0].object_id
+}
+
+resource "azurerm_role_assignment" "vdc_batch_worker_subnet" {
+  scope                = azurerm_subnet.batch_worker_subnet.id
+  role_definition_name = "Network Contributor"
+  principal_id         = azurerm_kubernetes_cluster.vdc.identity[0].principal_id
+}
+
+resource "azurerm_role_assignment" "vdc_k8s_subnet" {
+  scope                = azurerm_subnet.k8s_subnet.id
+  role_definition_name = "Network Contributor"
```

Comment:
I'm not sure what this block is for.

Reply:
See above comment. It kind of surprised me that I had to do this explicitly, but I see no harm in it because it is literally a subnet just for k8s.